cdc/cdc_partitioner.cc

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2020 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "cdc_partitioner.hh"
+#include "dht/token.hh"
+#include "schema.hh"
+#include "sstables/key.hh"
+#include "utils/class_registrator.hh"
+#include "cdc/generation.hh"
+#include "keys.hh"
+
+static const sstring cdc_partitioner_name = "com.scylladb.dht.CDCPartitioner";
+
+namespace cdc {
+
+const sstring cdc_partitioner::name() const {
+    return cdc_partitioner_name;
+}
+
+dht::token
+cdc_partitioner::get_token(bytes key) const {
```

Comment:
This is suboptimal, moving bytes around is expensive since it has to check if it's using internal or external storage each time.

As a follow up, please change stream_id to store two int64_t instead of bytes, and change the stream_id constructor to accept a bytes_view (and check during construction that the size is 16). Then this can also use bytes_view.

Reply:
It would be suboptimal because then we would have to transform those two ints into blob for every CDC write. It's way more common than this streams generation. I will introduce a static function to stream_id than takes bytes_view and returns token and will use it here instead.

Reply:
done